src/zenml/zen_server/template_execution/runner_entrypoint_configuration.py

```diff
@@ -36,4 +34,9 @@ def run(self) -> None:
         stack = Client().active_stack
         assert deployment.stack and stack.id == deployment.stack.id
 
-        deploy_pipeline(deployment=deployment, stack=stack)
+        placeholder_run = get_placeholder_run(deployment_id=deployment.id)
```

Comment:
Just a little bit unsure about this part. Is the relationship between a `deployment_id` and a `placeholder_run` always 1-to-1?

Reply:
It's 1 to 0 actually, which is why this function returns an optional placeholder run.